scripts/AI/States/Patrol.gd

```diff
@@ -33,6 +33,8 @@ func enter(_msg := {}) -> void:
 		vision_sensor.can_see_target.connect(_on_see_target)
 	if !state_machine.navigation_agent.target_reached.is_connected(_on_target_reached):
 		state_machine.navigation_agent.target_reached.connect(_on_target_reached)
+	if !hearing_sensor.sound_heared.is_connected(_on_sound_heared):
```

Comment:
Same here for the GUI stuff

Reply:
I need to connect dynamicallly here but I saw that I forget to disconnect to the signal when quitting the state